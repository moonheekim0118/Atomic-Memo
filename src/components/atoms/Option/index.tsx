import React from 'react';
import styled from 'styled-components';

interface Props {
  children: React.ReactNode;
  dataName: string;
}

const Option = ({ children, dataName }: Props) => {
  return (
    <StyledOption data-name={dataName} className="select-option">
      {children}
    </StyledOption>
  );
};

const StyledOption = styled.li`
  background-color: #3399ff;
  padding: 0.7em;

  transition: background-color 0.5s ease;
  &:hover {
    background-color: #66ccff;
  }
`;
export default Option;
