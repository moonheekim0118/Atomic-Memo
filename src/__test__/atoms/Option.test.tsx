import 'jsdom-global/register';
import React from 'react';
import Option from '../../components/atoms/Option';
import { mount } from 'enzyme';

describe('<Option/>', () => {
  let container = null;

  it('renders correctly', () => {
    container = mount(<Option dataName="test">test</Option>);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
