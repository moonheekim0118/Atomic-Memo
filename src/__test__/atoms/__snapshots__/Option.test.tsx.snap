// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option/> matches snpashot 1`] = `"<li data-name=\\"test\\" class=\\"Option__StyledOption-sc-1ojvno1-0 bYrEZC select-option\\">test</li>"`;
